components/server/ee/src/billing/entitlement-service-chargebee.ts

```diff
@@ -115,7 +108,7 @@ export class EntitlementServiceChargebee implements EntitlementService {
             return "unlimited";
         }
 
-        const diffInMillis = new Date(cachedStatement.endDate).getTime() - new Date(date).getTime();
+        const diffInMillis = Math.max(0, new Date(cachedStatement.endDate).getTime() - new Date(date).getTime());
```

Comment:
From reading this up and down, I'm pretty sure this new issue stems from the date passed down the line from https://github.com/gitpod-io/gitpod/blob/20106df07bb532ef45159dafa0cab5c3a05ef5f4/components/server/ee/src/billing/entitlement-service.ts#L37, these kind of timing issues are really nasty as they may occur in different places here. Be aware this is all a chain of async and await, and the date is set before awaiting a couple of db queries, right?

Reply:
Yes, it does. But in this case I don't see a problem with that. This is more about making the "guessing" correct for more cases, and we should not care about what dates we have where, or where we pull "now" from.

Reply:
Nice catch. It took a while to understand the issue here: any small positive or negative minus a large negative equals positive, which leads to growing negative remaining hours. I think we've seen these odd results in the past, but could not pin point the cause.